### ISSUE TITLE
[RE: JoSS Review] Undocumented `pysam` warning for STDIN streaming

### ISSUE THREAD

**Issue by betteridiot** [akikuno/cstag-cli#5]
For the JoSS review at openjournals/joss-reviews/issues/6066

Please see https://github.com/betteridiot/joss_reviews/blob/master/cstag/joss_review.md for complete notes.

Warning from `pysam` when dealing with STDIN regarding missing index file.
* Given that this warning will always occur when creating a file from STDIN, it may be good to provide either explicit notice to end-users what the error means (i.e., there is no index file for the newly created file from STDIN), provide a parameter to suppress the warning, or just have the code fully [workaround](https://github.com/pysam-developers/pysam/issues/939#issuecomment-669016051) the error.

**Comment by jmarshall** [comment]
The spurious warning in question is not emitted by the current pysam release.

**Comment by betteridiot** [comment]
For transparency, the version that was installed was v0.19.1 whereas the most current version is v0.22.0

That may be true @jmarshall and I do not dispute that upgrading would remove this spurious warning.

That said, I used only the following command to make my environment for this review.

`conda create -n joss_cstag --no-default-packages --solver libmamba bioconda::cstag bioconda::cstag-cli`

I did not pin any versions here. Therefore, this is the environment that `conda` resolves when installing, which leads to the spurious warning. This is just me being thorough.

**Comment by akikuno** [comment]
@betteridiot

Thank you very much for pointing out the warning message regarding missing index file.
Also, I appreciate letting me know the workaround.  I've incorporated this into the script.

https://github.com/akikuno/cstag-cli/commit/50b34f4eee05df9e17064a85bb16e8cb191ba0b0

@jmarshall
Thank you for informing me that the current version of pysam does not generate this warning.
Following your advice, I have suppressed the warning message.

Thank you so much!

**Comment by jmarshall** [comment]
The point of my comment is that by updating your pysam requirement in _pyproject.toml_, you would not need to implement this old workaround (and you would benefit from more recent pysam bug fixes).

**Comment by akikuno** [comment]
@jmarshall,
Thank you for your advice. I am planning to update pysam requirement.

My use of pysam version 0.19.1 has been primarily due to a compatibility issue with bioconda. I have adhered to this version for conda installations because more recent versions of pysam were failing tests, which hindered their inclusion in bioconda. Specifically, the error message encountered was: `pysam 0.21.0 requires cython, which is not installed`.

Here is the yaml file from another package of mine that experienced this error:
https://github.com/bioconda/bioconda-recipes/blob/67fec9fe812eb6f6dfb4caed233fbd4d08b0bde3/recipes/dajin2/meta.yaml#L18

Despite having added cython to the host, I encountered the error, so I was unable to determine the fundamental cause. However, by downgrading pysam and adjusting the versions of other dependencies, I was able to make the bioconda tests pass. Consequently, I have maintained this workaround.
https://github.com/bioconda/bioconda-recipes/pull/42741/files#diff-093bb56ca87be99bd8f948a62ffbd9e1f6fcf30902a6bdb4f38636365d68148aR31

Since my last attempt was three months ago, the situation may have changed. 
Therefore, I am considering updating the bioconda yaml file in conjunction with the next version upgrade of my package.

**Comment by jmarshall** [comment]
> pysam 0.21.0 requires cython, which is not installed.

This was an issue unfortunately introduced in 0.21.0, and is limited to that version only. It has been fixed in 0.22.0.

**Comment by akikuno** [comment]
Thank you very much for your guidance!

I have updated the Bioconda recipe, enabling the installation of the latest version of pysam. The update was completed smoothly, without any errors.
https://github.com/bioconda/bioconda-recipes/pull/44454/files

As a result, I have removed the workaround that was suppressing warnings.
https://github.com/akikuno/cstag-cli/commit/c6810bd57685fff94037cc6f1cd6b87b3fe42d48

Thank you again!
